src/fairchem/core/models/escn/escn.py

```diff
@@ -338,7 +338,12 @@ def forward(self, data):
         ###############################################################
         node_energy = self.energy_block(x_pt)
         energy = torch.zeros(len(data.natoms), device=device)
-        energy.index_add_(0, data.batch, node_energy.view(-1))
+        # CPU AMP does not allow this, but GPU AMP does. force conversion to float before hand
+        if node_energy.device.type == "cuda":
+            energy.index_add_(0, data.batch, node_energy.view(-1))
+        else:
+            energy.index_add_(0, data.batch, node_energy.float().view(-1))
```

Comment:
do we loose a lot of performance or use up too much memory when casting? if not then should we cast regardless?